### DISCUSSION TITLE
In Game Live Statistics Display

### DISCUSSION THREAD

**Discussion by LanTenDragon** [ppy/osu#13048] [Ideas]
**Describe the new feature:** 

Main Feature Request 

> Show the Unstable Rate of the current map in real-time. 

Score posts in Reddit almost always have UR. In stable, players have to hover over the performance grid (at the result screen) to see their UR. In lazer, there is no such grid. Therefore I would propose a live display shown in the proposal design below. Functionality wise, it can be extended from live PP counter

Secondary Feature Request 

> Allow the functionality to be extended to other statistics in the future like:

For all game modes 
>Unstable Rate
>Keys Per Second

osu!standard
>Bullseye / scatter plot of cursor position from circle center
>Mouse velocity  and acceleration

osu!mania
>Number of mispresses
>Keys Per Second Graph 

**Proposal designs of the feature:** 
![Untitled](https://user-images.githubusercontent.com/43511642/62851991-9fd75700-bd1a-11e9-9d68-1ec29041f6a0.png)

**Comment by Craftplacer** [comment]
I think it should include a live grade display, just to get a feel if you should be playing better.

**Comment by frenzibyte** [comment]
> I think it should include a live grade display, just to get a feel if you should be playing better.

Imho, the accuracy is more than enough for telling if you're playing good or not, you even see your estimated grade in breaks.

**Comment by Craftplacer** [comment]
> > I think it should include a live grade display, just to get a feel if you should be playing better.
> 
> Imho, the accuracy is more than enough for telling if you're playing good or not, you even see your estimated grade in breaks.

I have looked at the wiki detailing grades 1 to 2 times and only know arbitary numbers when each grade is passed, considering that grades can be represented using two letters or one icon I don't think it would hurt to include them.

**Comment by abraker95** [comment]
Would this be during gameplay or when viewing replay?

**Comment by LanTenDragon** [comment]
> Would this be during gameplay or when viewing replay?

Both. 

The player should choose whether to display both in gameplay and replay or only in replay or none at all. (Some players do not want anything on their screen while playing)

**Comment by Nouko61** [comment]
Cursor velocity looks cool, I tried to calculate the moving distance of my pen but its hard to be accurately. The software I used to record distance also makes my cursor lag.

**Comment by AuracleTech** [comment]
My post was different as what this post propose.
It's similar yes but this is not a pp counter, but this implies real time activity too.
Related yet not the same. I read this post before making one.

**Comment by bdach** [comment]
It isn't strictly the same thing but this issue is for all types of real-time display during gameplay, so the scope of this can be extended to cover pp too.

**Comment by vernonlim** [comment]
Adding on to this for when it's inevitably worked on despite it technically being a different proposal (since it isn't live).
I would really like it if there could be an option to display map difficulty settings and metadata, like the title, difficulty name, BPM, OD, SR, mods used, etc.

Why? I feel it would be really useful for livestreamers or viewers of osu! gameplay, because I always find myself wondering what those are while watching, and I have to wait until the play ends to see it in the menu, or open up a link to the map to see firsthand. I don't know how many times I've been left wondering what the BPM of a map is and have to judge it myself through hearing due to not having a link to the map, or seeing people in livestream chat asking whether a map is at AR11, or with HR. 

I think there should be enough space near the edge of the screen to fit these for all gamemodes, with a good design.
While the gameplay layout is obviously different, here's an example of it displayed in etterna (from a youtube video)
![image](https://user-images.githubusercontent.com/25032854/97447958-16d22d00-196b-11eb-8b7c-c6cd9c5cd7e0.png)
Map stats like difficulty rating, difficulty name, scroll speed and mods are at the bottom left, BPM is at the bottom middle, and song title is at the top. As a viewer, I know exactly what the streamer is playing.

**Comment by LeNitrous** [comment]
I do believe this can be merged with this issue: https://github.com/ppy/osu/issues/4113 since this issue proposes to expose the game state at least this offers a more better solution for streamers.

**Comment by enoslayd** [comment]
Im thinking that beatmap details can be displayed when in break time

**Comment by vernonlim** [comment]
> Im thinking that beatmap details can be displayed when in break time

definitely should be an option, but having (EDIT: only) it would be an issue for maps with no breaks, or short clips. in those cases, viewers still wouldn't know what the map settings are.

EDIT: I should add on despite it being obvious: exposing gameplay stats through an API is definitely needed as well — this is not a replacement for it at all and vice versa, since some content creators might need access to those stats to run their own plugins/programs not available in base osu (to add to a stream or video overlay, or to control other things), and other content creators might not want to go through the hassle of working with a plugin just to show viewers gameplay stats.
